scaling_performance/managing_hugepages.adoc

```diff
@@ -0,0 +1,87 @@
+[[scaling-performance-managing-huge-pages]]
+= Managing Huge Pages
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+
+== Introduction
+
+[IMPORTANT]
+====
+Managing huge pages is a Technology Preview feature only.
+ifdef::openshift-enterprise[]
+Technology Preview features are not supported with Red Hat production service
+level agreements (SLAs), might not be functionally complete, and Red Hat does
+not recommend to use them for production. These features provide early access to
+upcoming product features, enabling customers to test functionality and provide
+feedback during the development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+In {product-title}, applications in a pod can allocate and consume pre-allocated
+huge pages. This topic describes how.
+
+[[huge-pages-prerequisites]]
+== Prerequisites
+
+. Nodes must pre-allocate huge pages in order for the node to report its huge
+ page capacity. A node can only pre-allocate huge pages for a single size.
+
+. `HugePages` must be set to `true` across the system:
++
+----
+--feature-gates=HugePages=true
+----
+
+[[consuming-huge-pages]]
+== Consuming Huge Pages
+
+Huge pages can be consumed via container level resource requirements using the
+resource name `hugepages-<size>`, where size is the most compact binary
+notation using integer values supported on a particular node. For example, if a
+node supports 2048KiB page sizes, it will expose a schedulable resource
+`hugepages-2Mi`. Unlike CPU or memory, huge pages do not support overcommitment.
+
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  generateName: hugepages-volume-
+spec:
+  containers:
+  - image: fedora:latest
+    command:
+    - sleep
+    - inf
+    name: example
+    volumeMounts:
+    - mountPath: /hugepages
+      name: hugepage
+    resources:
+      limits:
+        hugepages-2Mi: 100Mi
```

Comment:
I think we need a call-out box related to this line.  Users of hugepages are used to specifying the amount of memory for hugepages a quantity of hugepages, multiplied by the size of the page.  They do this intuitively and it's been around for decades.

What we are doing here is different. We are allowing the user to specify the amount of memory for hugepages as the exact amount to be allocated.  So there is no "multiplied by the size of the page" step mental calculation.

In other words, given a hugepage size of 2MB, if a user wanted to use 100MB of hugepage-backed RAM for their application, they would allocate 50 hugepages.  Above you see we handle the math for users, allowing them to specify 100MB directly.